src/mapping/toroidal_to_cylindrical.hpp

```diff
@@ -0,0 +1,442 @@
+                          .deriv_1_and_2(coord, get_const_field(m_Z_spline_representation));
+        ddcHelper::get<Z, Phi_cov>(J) = 0.;
+        ddcHelper::get<Zeta, Rho_cov>(J) = 0.;
+        ddcHelper::get<Zeta, Theta_cov>(J) = 0.;
+        double Rmajor = m_spline_evaluator(coord, get_const_field(m_R_spline_representation));
+        ddcHelper::get<Zeta, Phi_cov>(J) = Rmajor * Rmajor;
+        return J;
+    }
+
+    /**
+     * @brief Get the index range describing the points which should be used to evaluate functions at the central point.
+     *
+     * @return An index range covering the O-point.
+     */
+    KOKKOS_INLINE_FUNCTION IdxRangeRhoTheta idx_range_singular_point() const
+    {
+        return m_idx_range_singular_point;
+    }
+
+    /**
+     * @brief Get a control point of the mapping on B-splines.
+     *
+     * The mapping @f$ (\rho,\theta) \mapsto (R,Z) @f$ decomposed on B-splines can be
+     * identified by its control points @f$ \{(c_{x,k}, c_{y,k})\}_{k}  @f$ where
+     * @f$ c_{x,k} @f$ and @f$ c_{y,k} @f$ are the B-splines coefficients:
+     *
+     * @f$ R(\rho,\theta) = \sum_{k=0}^{N_\rho\times N_{\theta}-1} c_{x, k} B_k{\rho,\theta} @f$,
+     *
+     * @f$ Z(\rho,\theta) = \sum_{k=0}^{N_\rho\times N_{\theta}-1} c_{y, k} B_k{\rho,\theta} @f$,
+     *
+     * where @f$ N_\rho\times N_{\theta} @f$ is the number of B-splines.
+     *
+     * The control points can be obtained by interpolating the mapping on interpolation
+     * points (see GrevilleInterpolationPoints or KnotsAsInterpolationPoints).
+     * We can also note that the first control points @f$ \{(c_{x,k}, c_{y,k})\}_{k=0}^{N_{\theta}} @f$
+     * are equal to the pole @f$ (c_{x,k}, c_{y,k}) = (R_0,Z_0) @f$, @f$ \forall k = 0, ..., N_{\theta}-1 @f$
+     * where @f$ R(0,\theta), Z(0,\theta) = (R_0,Z_0) @f$ @f$ \forall \theta @f$.
+     *
+     *
+     * @param[in] el
+     * 			The number of the control point.
+     *
+     * @return The el-th control point.
+     *
+     * @see GrevilleInterpolationPoints
+     * @see KnotsAsInterpolationPoints
+     */
+    KOKKOS_INLINE_FUNCTION const Coord<R, Z> control_point(
+            Idx<BSplineRho, BSplineTheta> const& el) const
+    {
+        return Coord<R, Z>(m_R_spline_representation(el), m_Z_spline_representation(el));
+    }
+};
+
+
+namespace mapping_detail {
+template <
+        class R,
+        class Z,
+        class SplineEvaluator,
+        class Rho,
+        class Theta,
+        class MemorySpace,
+        class ExecSpace>
+struct MappingAccessibility<
+        ExecSpace,
+        DiscreteToCartesian<R, Z, SplineEvaluator, Rho, Theta, MemorySpace>>
+{
+    static constexpr bool value = Kokkos::SpaceAccessibility<ExecSpace, MemorySpace>::accessible;
+};
+
+template <class R, class Z, class SplineEvaluator, class Rho, class Theta, class MemorySpace>
+struct IsCurvilinear2DMapping<DiscreteToCartesian<R, Z, SplineEvaluator, Rho, Theta, MemorySpace>>
+    : std::true_type
+{
+};
```

Comment:
This does not seem to be documented, but I think we use `IsCurvilinear2DMapping` to determine whether the following tags are defined:
```
cartesian_tag_x
cartesian_tag_y
curvilinear_tag_r
curvilinear_tag_theta
```

It certainly asks if the mapping is 2D which is not the case here